src/configuration.ts

```diff
@@ -124,8 +131,28 @@ export function loadConfigurationFromPath(configFilePath: string) {
         let fileData = fs.readFileSync(configFilePath, "utf8");
         // remove BOM if present
         fileData = fileData.replace(/^\uFEFF/, "");
-        return JSON.parse(fileData);
+        const configFile: IConfigFile = JSON.parse(fileData);
+        configFile.rulesDirectory = getRulesDirectories(configFile.rulesDirectory, path.dirname(configFilePath));
+        return configFile;
+    }
+}
+
+export function extendConfigFile(config: IConfigFile, baseConfig: IConfigFile): IConfigFile {
+    let combinedConfig: IConfigFile = {};
+
+    const baseRulesDirectory = arrayify(baseConfig.rulesDirectory);
+    const configRulesDirectory = arrayify(config.rulesDirectory);
+    combinedConfig.rulesDirectory = configRulesDirectory.concat(baseRulesDirectory);
+
+    combinedConfig.rules = {};
+    for (const name of Object.keys(objectify(baseConfig.rules))) {
+        combinedConfig.rules[name] = baseConfig.rules[name];
+    }
+    for (const name of Object.keys(objectify(config.rules))) {
+        combinedConfig.rules[name] = config.rules[name];
     }
+
+    return combinedConfig;
```

Comment:
Agree that `tslint` should not depends on `lodash` (`tslint` already take quite a while to install :worried:).
I meant to suggest creating a copy of `config` and apply/merge `baseConfig` using the `_.defaults` principle.